src/unix/lwt_unix_unix.h

```diff
@@ -1872,7 +1863,7 @@ CAMLprim value lwt_unix_readdir_n_job(value val_dir, value val_count)
     long count = Long_val(val_count);
     DIR *dir = DIR_Val(val_dir);
 
-    LWT_UNIX_INIT_JOB(job, readdir_n, dirent_size(dir) * count);
+    LWT_UNIX_INIT_JOB(job, readdir_n, sizeof(struct dirent) * count);
```

Comment:
It's good to get rid of the size estimation and whatnot. I believe this is not necessary with `readdir`, but see the other comment about copying out the strings.

For reference on how much of a pain this size estimation was, there is #292. Among other things, it fixed a buffer overflow bug in https://github.com/ocsigen/lwt/pull/292/commits/dbc42f26bd48873b3dbd864ab53e828d555f2111, where the janky `dirent` size computation recommended online could give a number smaller than what macOS could actually write, etc.

See also https://bugs.chromium.org/p/chromium/issues/detail?id=457759.

I sometimes wonder how anything in the world works, with such OS API issues :p Then again, that's why we have 0-days all over the place.